pyproject.toml

```
@@ -1,30 +1,88 @@
-
-[tool.coverage.run]
-branch = true
-source = ["sceptre"]
-
-[tool.coverage.report]
-exclude_lines = [
-  # Have to re-enable the standard pragma
-  "pragma: no cover", # Don't complain about missing debug-only code:
-  "def __repr__",
-  "if self.debug", # Don't complain if tests don't hit defensive assertion code:
-  "raise AssertionError",
-  "raise NotImplementedError", # Don't complain if non-runnable code isn't run:
-  "if 0:",
-  "if __name__ == .__main__.:", # Don't complain about mypy-specific code
-  "if TYPE_CHECKING:",
+[tool.poetry]
+name = "sceptre"
+version = "4.0.2"
+packages = [{ include = "sceptre" }]
+readme = "README.md"
+homepage = "https://github.com/Sceptre/sceptre"
+repository = "https://github.com/Sceptre/sceptre"
+documentation = "https://docs.sceptre-project.org"
+authors = ["Sceptre <sceptreorg@gmail.com>"]
+description = "An AWS Cloud Provisioning Tool"
+keywords = ["aws", "cloud", "devops", "infrastructure", "tools", "cli"]
+license = "Apache-2.0"
+classifiers = [
+  "Development Status :: 5 - Production/Stable",
+  "Intended Audience :: Developers",
+  "Natural Language :: English",
+  "Environment :: Console",
+  "Programming Language :: Python :: 3.8",
+  "Programming Language :: Python :: 3.9",
+  "Programming Language :: Python :: 3.10",
+  "Programming Language :: Python :: 3.11"
 ]
-ignore_errors = true
-fail_under = 90.0
-show_missing = true
-
-[tool.pytest.ini_options]
-# The "--cov" option will disable the pycharm debugger.  To enable the debugger either
-# remove this flag or add the "--no-cov" to your pycharm's pytest run configuration.
-# https://github.com/pytest-dev/pytest-cov/issues/131
-# https://stackoverflow.com/questions/34870962/how-to-debug-py-test-in-pycharm-when-coverage-is-enabled
-addopts = "-s --cov --cov-report term --cov-report term-missing --cov-report xml"
+
+[tool.poetry.scripts]
+"sceptre" = "sceptre.cli:cli"
+
+[tool.poetry.plugins."sceptre.hooks"]
+"asg_scheduled_actions" = "sceptre.hooks.asg_scaling_processes:ASGScalingProcesses"
+"cmd" = "sceptre.hooks.cmd:Cmd"
+
+[tool.poetry.plugins."sceptre.resolvers"]
+"environment_variable" = "sceptre.resolvers.environment_variable:EnvironmentVariable"
+"file_contents" = "sceptre.resolvers.file_contents:FileContents"
+"stack_output" = "sceptre.resolvers.stack_output:StackOutput"
+"stack_output_external" = "sceptre.resolvers.stack_output:StackOutputExternal"
+"no_value" = "sceptre.resolvers.no_value:NoValue"
+"select" = "sceptre.resolvers.select:Select"
+"stack_attr" = "sceptre.resolvers.stack_attr:StackAttr"
+"sub" = "sceptre.resolvers.sub:Sub"
+"split" = "sceptre.resolvers.split:Split"
+"join" = "sceptre.resolvers.join:Join"
+
+[tool.poetry.plugins."sceptre.template_handlers"]
+"file" = "sceptre.template_handlers.file:File"
+"s3" = "sceptre.template_handlers.s3:S3"
+"http" = "sceptre.template_handlers.http:Http"
+
+[tool.poetry.dependencies]
+python = ">=3.8,<3.12"
+boto3 = ">=1.3.0,<2"
+click = ">=7.0,<9.0"
+cfn-flip = ">=1.2.3,<2.0"
```

We can use the shortened syntax for this
```suggestion
cfn-flip = "^1.2.3"
```

---

done